### DISCUSSION TITLE
How to handle invalid input?

### DISCUSSION THREAD

**Discussion by vladdeSV** [jotaen/klog#162] [Feature ideas & discussions]
I read an article called "How to Read an RFC", and it took up an interesting point; what to do when some data does not meet the requirements laid out by the RFC.

Relevant section: https://www.mnot.net/blog/2018/07/31/read_rfc#interpreting-requirements

In quick summary: if I make my own klog parser, what shall my parse do when it encounters an invalid file?

I am not sure klog needs this, but I at least wanted to bring it up.

**Comment by jotaen** [comment]
Interesting point. The klog CLI fails altogether if anything in the input is not valid. I think that’s the most reasonable behaviour, and it probably wouldn’t harm to mention it in the spec.

[This statement in the spec](https://github.com/jotaen/klog/blob/main/Specification.md#ii-organising-records-in-files) probably comes closest to that topic currently:

> A file MAY hold any amount of records. Apart from that it MUST NOT contain anything but what is allowed by this specification.

If at all, it only covers the case indirectly, though. So maybe an additional sentence would be good to clarify this.
